src/providers/TokensProvider.tsx

```diff
@@ -42,15 +43,23 @@ export default function TokensProvider({
       }
       const ethCallProvider = await getMulticallProvider(library, chainId)
       const lpTokens = new Set()
+      const tokenType: { [tokenAddress: string]: PoolTypes | undefined } = {}
       const targetTokenAddresses = new Set(
         Object.values(basicPools)
           .map((pool) => {
             lpTokens.add(pool.lpToken)
-            return [
+            const tokensInPool = [
               ...pool.tokens,
               ...(pool.underlyingTokens || []),
               pool.lpToken,
             ]
+            Object.assign(
```

Comment:
I believe you could also just do `{ tokenType, ...tokensInPool.map((address) => ({ ...etc` thought I might be wrong on this.

Reply:
`...foo.map` would have array keys of 0,1,2 etc, so Object.assign is the way to go